dotnet-peer/Libp2pChat/Infrastructure/Libp2p/Libp2pChatService.cs

```diff
@@ -0,0 +1,342 @@
+            _logger.LogInformation("Chat service stopped");
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError("Error while stopping chat service", ex);
+            throw;
+        }
+    }
+    
+    /// <inheritdoc />
+    public void Dispose()
+    {
+        Dispose(true);
+        GC.SuppressFinalize(this);
+    }
+    
+    /// <summary>
+    /// Disposes resources.
+    /// </summary>
+    /// <param name="disposing">Whether to dispose managed resources.</param>
+    protected virtual void Dispose(bool disposing)
+    {
+        if (_isDisposed)
+            return;
+            
+        if (disposing)
+        {
+            _cancellationTokenSource?.Dispose();
+        }
+        
+        _isDisposed = true;
+    }
+    
+    private void HandleMessageReceived(byte[] data)
+    {
+        try
+        {
+            if (data == null || data.Length == 0)
+            {
+                _logger.LogWarning("Received empty message data");
+                return;
+            }
+            
+            string raw = Encoding.UTF8.GetString(data).Trim();
+            _logger.LogDebug($"Raw message received: {raw}");
+            
+            // Check if the message is JSON or plain text
+            bool isJsonMessage = raw.StartsWith("{");
+            if (isJsonMessage)
+            {
+                try
+                {
+                    // Try to deserialize as a ChatMessage
+                    var chatMessage = JsonSerializer.Deserialize<ChatMessage>(raw);
+                    if (chatMessage != null && !string.IsNullOrEmpty(chatMessage.Message))
+                    {
+                        // Don't process messages from ourselves
+                        if (chatMessage.SenderID == PeerId)
+                            return;
+                            
+                        // Track peer
+                        UpdateOrAddPeer(chatMessage.SenderID, chatMessage.SenderNick);
+                        
+                        // Raise event
+                        MessageReceived?.Invoke(this, chatMessage);
+                    }
+                    else
+                    {
+                        _logger.LogWarning($"Received invalid JSON message structure: {raw}");
+                    }
+                }
+                catch (JsonException ex)
+                {
+                    _logger.LogError($"JSON parsing error: {ex.Message}");
+                    
+                    // Treat as plain text
+                    var plainTextMessage = new ChatMessage(raw, "unknown", "peer");
+                    MessageReceived?.Invoke(this, plainTextMessage);
+                }
+            }
+            else
+            {
+                // Plain text message (likely from Go peer)
+                var plainTextMessage = new ChatMessage(raw, "unknown", "peer");
+                
+                // Add Go peer to known peers
+                var goPeer = Peer.CreateGoPeer();
```

Comment:
It can be any peer, go client is not a the only we expect to connect with. I'd suggest to change naming to something like peer or remotePeer